Include user journal in journal-gateway output

Samuel-BF opened the pull request:
When starting systemd-journal-gatewayd without directory option ( `-D`), only system journal are returned : 

```
$ logger -t testlog from-user
$ sudo logger -t testlog "from-sudo"
$ sudo journalctl SYSLOG_IDENTIFIER=testlog
-- Journal begins at Sun 2021-01-24 18:14:03 CET, ends at Sat 2021-04-24 18:42:02 CEST. --
avril 24 18:41:04 sid testlog[32339]: from-user
avril 24 18:41:58 sid testlog[32352]: from-sudo
$ curl "http://localhost:19531/entries?SYSLOG_IDENTIFIER=testlog"
Apr 24 18:41:58 sid testlog[32352]: from-sudo
```
And with `-D /var/log/journal` or with this patch :
```
$ curl "http://localhost:19531/entries?SYSLOG_IDENTIFIER=testlog"
Apr 24 18:41:04 sid testlog[32339]: from-user
Apr 24 18:41:58 sid testlog[32352]: from-sudo
```

The PR removes `SD_JOURNAL_SYSTEM` flag from `sd_journal_open`. The flag comes from the first version of systemd-journal-gatewayd ([7b17a7d72f5](https://github.com/systemd/systemd/commit/7b17a7d72f5ba5ad838b19803534c56a46f3bce9#diff-b4fbb92fd8d766bc254f9d0acea2220f9379c73d55e1e421e51ae219357d862bR91), 2012).

poettering commented:
Hmm, I think it would make sense to make this work like journalctl, i.e. accept `--user` and `--system` or a combination and then set the flags. 

I.e. I'd prefer if we would make this explicitly configurable, and not blindly open this up without providing a way to lock this down again.

Samuel-BF commented:
Actually, it would be great to open up all the options of journalctl for journal-upload and journal-gatewayd :
- arbitrary output format
- filter on any values


But for now, I'll try to implement the --user and --system flags (and --merge !)

Samuel-BF commented:
> Hmm, I think it would make sense to make this work like journalctl, i.e. accept --user and --system or a combination and then set the flags.

Done, inspired by systemd-journal-upload code (but without the code for accessing container journals). Note that : 

- Combining `-D` with `--user` or `--system` return no entries. Is it expected ? (same behaviour for journalctl and systemd-journal-upload)
- When setting `--file`, `--user` and `--system` flags have no effect (all entries are shown, compare [L117 to L119](https://github.com/systemd/systemd/commit/1840f81d74f05bdaabe5d79e1c53dd12b581f5e1#diff-732cab1fcb807574656bfc1b912fe5e3b41ecba76bdbda6bfb550bf3587f4681R117)) . Is it expected ? (same behaviour for journalctl and systemd-journal-upload)
- This PR changes default behaviour : user journals are now served by default by systemd-journal-gatewayd. They can be removed by specifying `--system` flag, just like in `journalctl`.

poettering commented on src/journal-remote/journal-gatewayd.c:
```diff
@@ -110,9 +114,11 @@ static int open_journal(RequestMeta *m) {
+                return sd_journal_open_files(&m->journal, (const char**) arg_file, 0);
         else
-                return sd_journal_open(&m->journal, SD_JOURNAL_LOCAL_ONLY|SD_JOURNAL_SYSTEM);
+                return sd_journal_open(&m->journal, !arg_merge*SD_JOURNAL_LOCAL_ONLY + arg_journal_type);
```
`+` ?

`|` !

poettering commented:
looks pretty good, but the "+" to add in bits i weird...

and yeah we could filter more invalid cmdline switch combinations. It's an explosion of combinations, and we are not good refusing all invalid ones.

Samuel-BF commented:
> looks pretty good, but the "+" to add in bits i weird...

Right. I fixed it, and fixed it in systemd-journal-upload also (where I borrowed the code snippet).


For reference, `SD_JOURNAL_SYSTEM` and `SD_JOURNAL_CURRENT_USER` flags for `sd_journal_open_*` are used in [sd-journal.c/file_type_wanted](https://github.com/systemd/systemd/blob/88c2c8a0ba13de31061a22a352410c18ffacab9a/src/libsystemd/sd-journal/sd-journal.c#L1210) to select journal files based on their name. That's why using these flags with `sd_journal_open_directory` makes sense while in `sd_journal_open_files` it's pointless.
